builtin/logical/transit/path_decrypt.go

```diff
@@ -134,6 +138,7 @@ func (b *backend) pathDecryptWrite(ctx context.Context, req *logical.Request, d
 				batchResponseItems[i].Error = err.Error()
 				continue
 			default:
+				p.Unlock()
```

Comment:
Why not just defer this on line 128 instead? Would probably be more bulletproof for future maintainers.

Reply:
Agree. Or is this part of the defer purge?

Reply:
Defer purge. Transit is one of the most perf sensitive backends we have.

Reply:
I finally caught up to this conversation https://github.com/hashicorp/vault/pull/4733#issuecomment-396273506 after I did this review. I don't know, I'm not totally sold yet on stripping defers. While it might provide better speed, my concern is we'd spend more time tracking future bugs from higher complexity. I think it depends on whether the benefit would be perceptible to users, and useful enough to justify the trade-off.